systemtest/approvals.go

```diff
@@ -0,0 +1,171 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package systemtest
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/tidwall/gjson"
+	"github.com/tidwall/sjson"
+
+	"github.com/elastic/apm-server/systemtest/estest"
+)
+
+const (
+	// approvedSuffix signals a file has been reviewed and approved
+	approvedSuffix = ".approved.json"
+
+	// receivedSuffix signals a file has changed and not yet been approved
+	receivedSuffix = ".received.json"
+)
+
+// ApproveEvents compares the _source of the search hits with
+// the contents of the file in "systemtest/approvals/<name>".
+//
+// Dynamic fields (@timestamp, observer.id, etc.) are replaced
+// with a static string for comparison. Integration tests elsewhere
+// use canned data to test fields that we do not cover here.
+//
+// If the events differ, then the test will fail.
+func ApproveEvents(t testing.TB, name string, hits []estest.SearchHit) {
```

Comment:
Why are you not using the existing approvals code?

Reply:
I did start going down that route. I ended up creating a new thing in order to keep a clear separation between the systemtest module and the main apm-server module.

Reply:
Discussed live: I'll look into creating another module in the repo which provides a shared package.